### PR TITLE
Volto docs integration

### PR DESCRIPTION
**Pull request by tiberiuichim**
@stevepiercy A bit late, but here it is. 

I've also added a Volto PR to match it. We're using that branch right now in the submodule, but that can be configured.

I've also played a bit with the Makefile, to try to make it more "plug and play", for example to avoid the virtualenv activation step.

Let me know if there's issues and I'll try to fix. Cheers!

### CODE REVIEW

**Comment by stevepiercy** [comment]
w00t! It works! This is great!

We need to make a few docs updates to https://github.com/plone/documentation/edit/6-dev/docs/contributing/setup-build.md, and also update https://github.com/plone/documentation/blob/b37d36d7d7253fa6aea5bdd9c1641e1d3a1141de/.github/workflows/build_deploy.yml. I would also like @ksuess to give it a try.

**Comment by sneridagh** [comment]
I gave it a try and LGTM!

Should we change from mkdocs to a MyST build in Volto as well? What do you think? I'd say so, to maintain coherence before all is settled. During the transition, it would be to maintain docs.voltocms.com up and running using the MyST build.

We will need to fix links and spelling, to keep the tests happy, right? I'm happy to help too.

**Comment by tkimnguyen** [comment]
Point me at whatever you want me to review or tweak or edit!

**Comment by tiberiuichim** [comment]
@tkimnguyen If you get this branch built, it uses https://github.com/plone/volto/pull/3001 as the linked "docs source", so any changes to Volto docs need to be done in the docs/volto subfolder and committed and pushed from the submodules/volto subfolder. When you make changes to the volto submodule, also make a commit in the main folder, to update the linked commit pointer for the submodule.

**Comment by tiberiuichim** [comment]
@sneridagh It makes sense to convert volto docs to MyST, to keep things smooth.

**Comment by stevepiercy** [comment]
If the docs are in CommonMark flavor of Markdown, there is no need to convert. MyST is a superset of CommonMark. If they have extensions, those parts are not supported and will need to be converted manually.

Also if you want to add reST or Sphinx features, such as cross-reference links, Intersphinx, admonitions, Glossary, and so on, then you can start adding them to the Volto docs.

**Comment by stevepiercy** [comment]
We also need to add instructions for contributors to go to the source repo to edit volto docs, because they will expect to find them here. Especially the exact commands involved with submodules. I can never remember the `git submodule` commands.

There are also issues with the GitHub buttons on the deployed site with two repos. We'll continue to point to this repo, but we'll need to transfer Volto docs issues to the Volto repo as they come in. Oh, and I need to fix the branch it currently points to.

File: .gitignore
Comment (stevepiercy) [inline .gitignore]: I do not see where either of these are generated. Either please explain or put into `~/.gitignore_global` and remove.

File: Makefile
Comment (stevepiercy) [inline Makefile]: Is this required? I use zsh.

File: Makefile
Comment (stevepiercy) [inline Makefile]: Please remove. This prevents using just `make` on the command line to output the list of available make targets and their comments.

File: Makefile
Comment (stevepiercy) [inline Makefile]: Please revert replacement of spaces with tabs for all comment indents.

File: Makefile
Comment (stevepiercy) [inline Makefile]: What is the purpose of this comment?

File: Makefile
Comment (stevepiercy) [inline Makefile]: I think this target is redundant to `livehtml`. I don't think we need this, unless I am missing something.

**Review by stevepiercy — CHANGES_REQUESTED** [review]
I have requested a few changes and have some questions for your consideration. Thank you!

File: Makefile
Comment (stevepiercy) [inline Makefile:39]: This target is the same as `build`, but with statement delimiters. These two targets should be merged into one. I prefer the cleaner syntax of `build` without statement delimiters, and prefer the target name of `bin/python`. Is there a reason for using statement delimiters?

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Comment by stevepiercy** [comment]
To keep this PR short, let's focus on integration, specifically how to pull in the Volto docs and build them in this repo, and how to make updates to them. Content and conversion to MyST (if needed) can be done in separate PRs.

Here's a list of remaining tasks, as far as I see. Please feel free to add or edit.

- [x] Resolve Review comments and questions.
- [x] Update instructions in `docs/contributing/setup-build.md`, per resolution of Review.
- [x] Update `docs/contributing/setup-build.md` to reflect `make help` instead of just `make`.
- [x] Add section for how to add a remote repo to the docs in `docs/contributing/authors.md`. See https://github.com/plone/documentation/pull/1160#discussion_r790799617
- [x] Add instructions of how to update Volto docs, then pull those updates into this repo, in `docs/contributing/index.md`.
- [x] Add instructions for Volto contributions under the section `Making contributions on GitHub` in `docs/contributing/index.md`.

**Comment by stevepiercy** [comment]
> @tkimnguyen If you get this branch built, it uses [plone/volto#3001](https://github.com/plone/volto/pull/3001) as the linked "docs source", so any changes to Volto docs need to be done in the docs/volto subfolder and committed and pushed from the submodules/volto subfolder. When you make changes to the volto submodule, also make a commit in the main folder, to update the linked commit pointer for the submodule.

@tiberiuichim can you put this into commands? Also what do you consider to be the "main folder"? Finally are the changes supposed to be made in this repo or volto's?

FWIW, I edited a volto file in this repo, pushed the commit to this branch/PR, but it does not go upstream to the volto repo. We want a process where changes are pushed to the upstream volto repo. Do developers need to add a git remote?

Please let us know. Thank you!

File: Makefile
Comment (tiberiuichim) [reply Makefile]: I've added it as a way to test the generated static html files. It was unclear to me how Sphinx generates the toctree based navigation and why it missed, sometimes, the section that I was adding. I can remove it, but I find it reassuring to have an easy way to "test the final build".

File: Makefile
Comment (tiberiuichim) [reply Makefile]: I can remove. Just as a reminder of how the submodule was added.

File: Makefile
Comment (tiberiuichim) [reply Makefile]: Makefile targets list their dependencies after the colon, without some spacing it might look confusing, so I typically use two tabs to separate the help comment from the rest of the target definition. I will remove the tabs.

File: Makefile
Comment (tiberiuichim) [reply Makefile]: Ok, I will set "help" as the default goal. The problem is that makefiles use the first defined target when you simply run `make`, so it can be confusing if people aren't familiar with makefiles.

File: Makefile
Comment (tiberiuichim) [reply Makefile]: I'll remove it, but even if you use zsh (I use fish), it's only for the variable handling internal to the processing of this Makefile. See https://www.gnu.org/software/make/manual/html_node/Choosing-the-Shell.html. I've added it thinking that it may be needed for the realpath part.

File: .gitignore
Comment (tiberiuichim) [reply .gitignore]: I have them in my .gitignore_global, for some reason they get ignored. I will remove them.

**Comment by tiberiuichim** [comment]
@stevepiercy Here's my workflow to edit the Volto docs:

```
cd docs/volto
edit + save any docs/volto/*.md
cd ../submodules/volto
git commit -am "Message intended for plone/volto repo"
git push
cd ../../
git commit submodules/volto -m "Updated submodule/volto tip"
git push
```

Basically, treat submodules/volto as a separate repo. Branch it, commit/push separately in it, etc. Then, once the plone/volto repo (let's say master, but for now is the plone6docs branch) has the documentation updates in it, make sure to update the local copy to that HEAD and then go to the plone/documentation folder and make a commit with the SHA of the module. I think the key to understanding is looking at this page, https://github.com/plone/documentation/tree/volto_docs/submodules notice that the volto link points to a particular commit. Basically, we need a separate commit inside "documentation" to update what the submodule commit tip should be.

At some point we will want to update the submodule, and that can be done with `git submodule update`.

File: Makefile
Comment (stevepiercy) [reply Makefile]: livehtml does that and more.

Sometimes you will need to do a `make clean`, as Sphinx only updates files that have changed and not the navigation site-wide. I do that when I edit `toctree` as it is confusing.

There are some things that you need to be aware of, but if you follow the pattern established at each level of docs (`docs/index.md` and `docs/classic-ui/index.md` for example), then the navigation should be happy. Feel free to ping me on Discord.

File: Makefile
Comment (stevepiercy) [reply Makefile]: That is what I assumed. Let's get it documented in `docs/contributing/authors.md`, as it is one of those things that is hard to remember, and we might need to do this for other repos (maybe restapi?). I've added this as a task

File: Makefile
Comment (stevepiercy) [reply Makefile]: Thank you. Two spaces is good enough, is familiar to Python inline comments after a statement, and what we've already done with Trainings.

File: Makefile
Comment (stevepiercy) [reply Makefile]: I think you are right. `make help` is more understandable and would be a good default target for just `make`.

We will also need to update the Contributing docs to use `make help` instead of just `make` to get the list of targets and their comments. Explicit is better than implicit.

File: Makefile
Comment (stevepiercy) [reply Makefile]: Thank you for the reference. After reading that link, I think keeping the `SHELL` var is a good idea.

**Comment by ksuess** [comment]
I can build. 
I can modify a Volto doc [0] and see the change in the build with `make livehtml`.
I did not risked to make a submodule update to commit to https://github.com/plone/volto/. Is this already in place? I do not want to interfere with ongoing work. Can I test and / or help?

[0]

````
```{note}
The `Listing` block already supports several of them (only in the "template" or…
```
````

**Review by ksuess — APPROVED** [review]

**Comment by tiberiuichim** [comment]
@ksuess You can use the following workflow (in generic terms):

- Right now we're using the plone6-docs branch from plone/volto. This is linked as a submodule in plone/documentation.
- When you pull this plone/documentation branch, it will set the submodules/volto folder to a particular commit, which you can see here: https://github.com/plone/documentation/tree/volto_docs/submodules. From git's point of view, the submodules/volto is on a "detached head", as it doesn't track a particular branch. So you need to run:

```
cd submodules/volto
git checkout plone6-docs
```
Now you can edit the files in docs/volto, they're symlinked from submodules/volto. When you want to make a commit, go to the submodules/volto and just do a commit there, with:

```
git add changed/files
git commit -m "..."
```

Once that's done, you need to go also to the root folder of this repo, and do another commit there:

```
cd ../../
git add submodules/volto
git commit -m "Updated submodules/volto tip"
```

And that is all. If I'm a contributor and I pull from the repo, my subfolders/volto repo will be updated to the same tip that you have commited for submodules/volto.

I have discussed with Steve and, for the future (at least for the immediate future) we have the following principles:

- Volto documentation will continue to be updated in plone/volto (and @stevepiercy will try to get a Myst-based setup for plone/volto)
- Whenever there's a Volto release, we can have a process to update the tip of the submodules/volto folder to that release, so that we track Volto releases.
- There will be a bit of learning curve now, in the beginning, while we adjust to the submodules workflow. In the future there will be less friction, I would say, as the Volto documentation is directly maintained by the Volto team, so only the Documentation team will not do a lot of work there (except at this stage, while we bring everything in line). Reusing the plone/documentation Sphinx stack in plone/volto will ensure that we keep things compatible and smooth running.

@sneridagh FYI ^

File: Makefile
Comment (tiberiuichim) [reply Makefile]: In the latest version of the Makefile "help" is now the default target. I think that's the best. Usually "all" is the default target in makefile, when I've set that it was by force of habit.

**Comment by stevepiercy** [comment]
I added documentation for how to update volto docs with submodules in https://github.com/plone/documentation/pull/1160/files#diff-1aeecae8f389487f8c42feed5ae17f247adf21aea14e59918c3b6ac7f8d1b033

I also ran through that process, using https://github.com/plone/volto/pull/3017 as an example, and it seems to have worked for me.

I would appreciate someone else running through it and try to break it or improve it. @tkimnguyen @ksuess @tiberiuichim @sneridagh

File: .gitignore
Comment (stevepiercy) [reply .gitignore]: Fixed in 48b5ab20019214715481ba3e7f3dbc5f27affe22

File: Makefile
Comment (stevepiercy) [reply Makefile]: This was resolved in e29c54136c1e9d75811c7a3fb2b344f10b635eaf

File: Makefile
Comment (stevepiercy) [reply Makefile]: Fixed in 3ee76c9f93654dbbe9505b70c6054ccf9ca30797

File: Makefile
Comment (stevepiercy) [reply Makefile]: This was fixed in a couple of commits 042ad3327316f4a3a7932eb489da4ebf515a5e6b and 024c8fee982632e0998dce17d70a91bf50d53e2b

File: Makefile
Comment (stevepiercy) [reply Makefile]: @tiberiuichim please check the commit 649811f0d0ddabf7f918e234c6d76e5fc37415a0 to make sure I understood how to do this.

File: Makefile
Comment (stevepiercy) [reply Makefile]: Fixed in 3ee76c9f93654dbbe9505b70c6054ccf9ca30797

File: Makefile
Comment (stevepiercy) [reply Makefile:39]: Fixed in 3ee76c9f93654dbbe9505b70c6054ccf9ca30797

**Comment by stevepiercy** [comment]
I completed my [above checklist](https://github.com/plone/documentation/pull/1160#issuecomment-1019925157), except for a review from @tiberiuichim in https://github.com/plone/documentation/pull/1160#discussion_r795311314

**Review by nileshgulia1 — APPROVED** [review]
I gave it a try, LGTM.

**Comment by stevepiercy** [comment]
@nileshgulia1 thaks for the verification!

In the spirit of "merge first, ask questions later", keep momentum, and to unblock backend and classic-ui from contributing, I will merge this PR by end of day Tue, Feb 1. I would still like one more reviewer before merging. Always good to have another pair of eyes. Thank you!

**Comment by ksuess** [comment]
I am puzzled and need an update. If we review /plone/documentation/pull/1160 then /plone/volto/tree/plone6-docs is pulled in. is /plone/volto/tree/plone6-docs the branch we should look at (it seems to be the one that's adjusted to new Plone 6 docs, but not up to date with master as it was not freezed, and naturally not mergable anymore in Volto with no mkdocs artifacts)

**Comment by ksuess** [comment]
I think http://127.0.0.1:8000/contributing/authors.html#importing-external-docs-with-submodules is not a topic for authors but for documentation administrators.

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md]:
```suggestion
### Working with the `plone/documentation` repository
```
- explicit mention of plone/documentation in contrary to plone/volto/docs
- request to native speakers: where and if to put the  word 'only'?

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md]:
```suggestion
This section describes how to make contributions to files in the `plone/documentation` repository only, and excludes `plone/volto/docs` files.
```

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md]:
```suggestion
1.  From the project root directory, sync your local `6-dev` branch with its remote.
```

**Review by sneridagh — APPROVED** [review]

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md:159]:
```suggestion
1.  After the build pass, commit changes to your branch, and push it to the remote repository on GitHub.
```

**Review by ksuess — CHANGES_REQUESTED** [review]

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md]:
```suggestion
If you want to edit the Volto documentation: files within the `docs/volto` subdirectory, the process is slightly different.
```

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md]:
```suggestion
### Editing Volto documentation

Working with both the `documentation` and `volto` repositories
```

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md]: It is not described in the mentioned section.

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md]:
```suggestion
We currently use the branches `plone/documentation@6-dev` and `plone/volto@plone6-docs` as the main branches for developing Plone 6 Documentation.
```

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md]: native speaker: isn't it sync?

**Review by ksuess — CHANGES_REQUESTED** [review]

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md:231]:
```suggestion
1.  Now return to the project root directory, and update the submodule to point to the commit you just made, and push your changes to the remote /plone/documentation repository.
```

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md]: This link is OK. But a link to a compare with the branch would be even better. Anyone expert on this field?

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md]:
```suggestion
1.  Members who subscribe to the plone/volto repository will receive a notification and review your request.
```

**Review by ksuess — CHANGES_REQUESTED** [review]

File: docs/contributing/index.md
Comment (ksuess) [inline docs/contributing/index.md:222]: Add a changelog entry.

why not introduce  ## Documentation section in plone/volto?

```
### Documentation

- Add metadata on i18n page @janedoe
```

**Review by ksuess — CHANGES_REQUESTED** [review]

**Comment by ksuess** [comment]
Still missing: 
- Merge of PR on plone/volto (if it affects plone/volto/docs)  triggers a rebuild of docs.plone.org

File: Makefile
Comment (stevepiercy) [reply Makefile]: Resolving per subsequent reviews from @ksuess and @nileshgulia1

**Comment by stevepiercy** [comment]
> I think http://127.0.0.1:8000/contributing/authors.html#importing-external-docs-with-submodules is not a topic for authors but for documentation administrators.

Yeah, this is not a great place for this content. What do you think of adding a page `contributing/admin.md` for this content and potential other future admin things?

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md]: Fixed in e8d1338865d54b825addfe54761b10285d866628

The intent is to say that the contributor is working with only this repo and no other.

"Only" is a fascinating English word. It gets tossed around for emphasis and meaning. It's hard to explain. Here's a good example.

https://englishwithnab.com/place-the-word-only-anywhere-on-the-sentence/

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md]: I am showing my age. Trends over the last two decades show that "sync" is now the more common usage. Reference: https://grammarhow.com/sync-vs-synch/

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md:159]: There are multiple builds that must pass, not just one.

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md]: There is an English phrase, "It's not 'if', but 'when`," meaning that something is inevitable. You're right, it is not inevitable, but it is from only my perspective 😆 .

Fixed in 32fb1ba43814949f76148cc9e6cafee726b8834a and 38c9510bd181045bee7592dec577fcf38e926ef4

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md]: Fixed in 89e81f9cddbd899687c0d9eaf2e6118963d076c4

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md]: @ksuess I rewrote this passage. Also see in `setup-build`:

> Then with a single command using `Makefile`, create a Python virtual environment, install project dependencies, **pull in Volto documentation as a git submodule**, build the docs, and view the results in a web browser by opening `/_build/html/index.html`.

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md]: Fixed in e45ae4f067067f407d894cd72efce49ed5b50f62

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md:231]: The remote branch could be in either a user's fork or the upstream repo `/plone/documentation`. Not being explicit allows either usage.

**Comment by ksuess** [comment]
> > I think http://127.0.0.1:8000/contributing/authors.html#importing-external-docs-with-submodules is not a topic for authors but for documentation administrators.
> 
> Yeah, this is not a great place for this content. What do you think of adding a page `contributing/admin.md` for this content and potential other future admin things?

yes, sounds fine.

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md]: Yes, I tried, but it is a chicken-and-egg thing. You cannot create a link to compare a branch that does not exist. This is the best default option I could come up with.

Anyway, here is an example of comparing `master` as the base against `plone6-docs`: https://github.com/plone/volto/compare/master...plone6-docs

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md]: Fixed in e06cdbb891c7417b07a076bd0d48c38724c0d160

File: docs/contributing/index.md
Comment (stevepiercy) [reply docs/contributing/index.md:222]: Agreed. I'll just move things around, and if anyone complains, slap me with a fish.

**Comment by stevepiercy** [comment]
@ksuess I am done with the changes. This is ready for a final review. Thank you!

**Review by ksuess — APPROVED** [review]
Awesome! 
Maybe just integrating your new admistrators guide in toc/navigation.

**Comment by polyester** [comment]
Did drive-by testing now on this branch, Ubuntu 20.04. It builds, live-html, etc. Looks good to me.

 \o/

**Comment by stevepiercy** [comment]
> Awesome! Maybe just integrating your new admistrators guide in toc/navigation.

Thank you, and fixed.

Also found and squashed a couple of other little bugs.

**Comment by stevepiercy** [comment]
Netlify preview builds work again.

The test failures can be ignored for now, as we are still working on infrastructure.

This is now ready for merge.

Let's see how it goes with https://github.com/plone/buildout.coredev/issues/768#issuecomment-1020578401

**Pull request merged**
